### ISSUE TITLE
Sync crosshair/vertical cursor across charts

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#83]
**Is your feature request related to a problem? Please describe.**
Multiple charts can share the same vertical cursor when they have the same x axis.

**Describe the solution you'd like**
We need to add two props like the following to the chart:
- `onExternalCursorChange` that is populated with the value from an external cursor
- `onCursorChange` that accept a listener and is called when the cursor change it's position

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Kibana Cross Issues**
Add any [Kibana](https://github.com/elastic/kibana) related issues/feature requests here.

**Checklist**

- [x] this request is checked against already exist requests
- [ ] every related Kibana issue is listed under `Kibana Cross Issues` list
- [ ] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

**Comment by nickofthyme** [comment]
@markov00 what are your thoughts on limiting this functionality to time series only?

I need to trigger the event with the actual `xValue` (i.e. time) rather than just the cursor `Dimensions` becuase charts could be different widths on the dashboard.

It's not that hard to parse the `xValue` for all charts but just curious if there is a use case to sync the cursors accross non-time-series charts.

**Comment by markov00** [comment]
closed by #304
